plugins/outputs/prometheus_client/prometheus_client.go

```diff
@@ -78,6 +92,38 @@ var sampleConfig = `
   collectors_exclude = ["gocollector", "process"]
 `
 
+func (p *PrometheusClient) getTLSConfig() *tls.Config {
+	if p.TLSCert != "" && p.TLSKey != "" {
+		return &tls.Config{
+			MinVersion:               tls.VersionTLS12,
+			PreferServerCipherSuites: true,
+			CipherSuites: []uint16{
+				tls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
+			},
```

Comment:
Shouldn't we be using p.TLSCert and p.TLSKey somewhere?  Why are we so specific on the ciphers and tls version?

Reply:
We specify the certificate and key in [ListenAndServeTLS() in line 171](https://github.com/influxdata/telegraf/pull/3719/files#diff-7879f2f3a90e7b811db8139a478ef350R171).

The specified cipher is the [mandatory cipher for HTTP/2](https://tools.ietf.org/html/rfc7540#section-9.2.2) which is used by prometheus when using TLS. If you want we can leave it open but it is the mandatory cipher for HTTP/2.

Reply:
Thanks for the explanation, yeah lets leave CipherSuites and PreferServerCipherSuites unset so that it will use the defaults.